plugin-cgroupv1/src/oar2/plugin.rs

```diff
@@ -270,6 +134,104 @@ impl AlumetPlugin for Oar2Plugin {
     }
 }
 
+impl Oar2Plugin {
+    fn job_id_from_name(name: &str) -> Result<String, anyhow::Error> {
+        Ok(name.split_once('_').context("Invalid oar cgroup")?.1.parse()?)
+    }
+
+    //TODO: could implement here some configuration that would enable/disable some metrics collections (using filepaths Options)
+    fn new_job_source(
+        job_name: &String,
+        metrics: Metrics,
+        cpuacct_controller_path: &Path,
+        memory_controller_path: &Path,
+    ) -> Result<Box<OAR2Probe>, anyhow::Error> {
+        let job_id = Self::job_id_from_name(job_name)?;
+
+        let cpuacct_controller_job_path = cpuacct_controller_path.join(job_name);
+        log::debug!("CPU path {cpuacct_controller_job_path:?}");
+        let memory_controller_job_path = memory_controller_path.join(job_name);
+        log::debug!("Memory path {memory_controller_job_path:?}");
+
+        let cpuacct_usage_filepath = cpuacct_controller_job_path
+            .join("cpuacct.usage")
+            .to_str()
+            .unwrap()
+            .to_string();
+        let memory_usage_filepath = memory_controller_job_path
+            .join("memory.usage_in_bytes")
+            .to_str()
+            .unwrap()
+            .to_string();
+
+        Ok(Box::new(OAR2Probe::new(
+            job_id,
+            metrics,
+            Some(cpuacct_usage_filepath),
+            Some(memory_usage_filepath),
+        )?))
+    }
+}
+
+struct JobDetector {
+    cpuacct_controller_path: PathBuf,
+    memory_controller_path: PathBuf,
+    control_handle: PluginControlHandle,
+    metrics: Metrics,
+    trigger: TriggerSpec,
+    rt: tokio::runtime::Runtime,
+}
+
+impl EventHandler for JobDetector {
+    fn handle_event(&mut self, event: Result<Event, notify::Error>) {
+        fn handle_event_on_path(job_detect: &mut JobDetector, path: PathBuf) -> anyhow::Result<()> {
+            if let Some(job_name) = path.file_name() {
+                let job_name = job_name.to_str().unwrap().to_string();
+
+                if job_name.chars().any(|c| c.is_numeric()) {
+                    let source = Oar2Plugin::new_job_source(
+                        &job_name,
+                        job_detect.metrics.clone(),
+                        &job_detect.cpuacct_controller_path,
+                        &job_detect.memory_controller_path,
+                    )?;
+
+                    let source_name = job_name;
+                    let new_job_source =
```

Comment:
Optimization (maybe later): build a list of sources to create and use `request::create_many()`. You can put a TODO if you feel like it would be better in a follow up PR.